File: second.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int getBinary(int x, int n){
    int binary = 0;
    int multiple = 1;
    int temp = x;
    while(temp>0){      
        int digit = temp%2;
        binary = binary + (digit*multiple);
        multiple = multiple*10;
        temp = temp/2;
    }
    printf("The binary value for %d is %d\n", x, binary);
    int hold = n;
    while(hold>0){
        binary = binary/10;
        hold--;
    }
    return binary%(10);
}

int complement(int x, int n){
    int array[30];
    int temp = x;
    int index = 0;
    while(temp>0){
        int digit = temp%2;
        array[index] = digit;
        index++;
        temp = temp/2;
    }
    if(array[n] == 0){
        array[n] = 1;
    }else if(array[n] == 1){
        array[n] = 0;
    }
    int finale = 0;
    index-=1;
    while(index>-1){
        int power = 2;
        if(index == 0){
            power = 1;
        }else{
            for(int i = 1; i<index; i++){
                power = power*2;
            }
        }
        finale = finale + array[index] * power;
        index--;
    }
    return finale;
}

int set(int x, int n, int v){
    int array[30];
    int temp = x;
    int index = 0;
    while(temp>0){
        int digit = temp%2;
        array[index] = digit;
        index++;
        temp = temp/2;
    }
    array[n] = v;
    index-=1;
    int finale = 0;
    while(index>-1){
        int power = 2;
        if(index == 0){
            power = 1;
        }else{
            for(int i = 1; i<index; i++){
                power = power*2;
            }
        }
        finale = finale + array[index] * power;
        index--;
    }
    return finale;
}

int main(int argc, char *filename[argc+1]){
    //OPEN FILE
    FILE *file = fopen(filename[1], "r");
    if(file == NULL){
        EXIT_FAILURE;
    }
    //GET MANIPULATED VALUE
    int value;
    fscanf(file, "%d", &value);
    //GET FIRST OPERATION
    char operation[28];
    while(fscanf(file, "%s", operation) != EOF){
        if(strcmp(operation, "get") == 0){
            int temp1;
            int temp2;
            fscanf(file, "%d", &temp1);
            printf("%d\n", getBinary(value, temp1));
        }else if(strcmp(operation, "comp") == 0){
            int temp1;
            int temp2;
            fscanf(file, "%d", &temp1);
            fscanf(file, "%d", &temp2);
            value = complement(value, temp1);
            printf("%d\n", value);
        }else if(strcmp(operation, "set") == 0){
            int temp1;
            int temp2;
            fscanf(file, "%d", &temp1);
            fscanf(file, "%d", &temp2);
            value = set(value, temp1, temp2);
            printf("%d\n", value);
        } 
    }    
    EXIT_SUCCESS;
}

File: second.java
import java.io.FileNotFoundException;
import java.io.File;
import java.util.Scanner;

public class second {
    public static int set(int manipulatedValue, int n, int v){
        int[] array = new int[30];
        int temp = manipulatedValue;
        int index = 0;
        while(temp>0){
            int digit = temp%2;
            array[index] = digit;
            index++;
            temp = temp/2;
        }
        array[n] = v;
        for(int val : array){
            System.out.println(val);
        }
        int numOfDigits = index;
        int finale = 0;
        int nIndex = 0;
        while(numOfDigits != 0){
            finale = finale + array[nIndex]*(int)(Math.pow(2, index));
        }
        for(int i = 0; i==numOfDigits; i++){
            finale = finale + array[i]*(int)Math.pow(2, index);
            index--;
        }
        return finale;
    }
    public static int getBinary(int manipulatedValue, int n){
        int binary = 0;
        int multiple = 1;
        int temp = manipulatedValue;
        while(temp>0){
            int digit = temp%2;
            binary = binary + (digit*multiple);
            multiple = multiple*10;
            temp = temp/2;
        }
        System.out.println("The binary value is: "+binary);
        int hold = n;
        while(hold>0){
            binary = binary/10;
            hold--;
        }
        System.out.println("The correct digit is: "+binary%10);
        return binary%(10);
    }
    public static int comp(int manipulatedValue, int n){
        int[] array = new int[30];
        int temp = manipulatedValue;
        int index = 0;
        while(temp>0){
            int digit = temp%2;
            array[index] = digit;
            index++;
            temp = temp/2;
        }
        //NOW CHANGING THE VALUE TO COMPLEMENT
        if(array[n] == 0){
            array[n] = 1;
        }else if(array[n] == 1){
            array[n] = 0;
        }
        int finale = 0;
        index-=1;
        while(index>=0){
            finale = array[index]*(int)(Math.pow(2, index));
            index--;
        }
        return finale;
    }
    public static void main(String[] args) {
         try{
            File read = new File("second.txt");
            Scanner input = new Scanner(read);
            int manipulatedValue = input.nextInt();
            String[] line = input.nextLine().split(" ");
            if(line[0].equals("get")){
                get(manipulatedValue, Integer.parseInt(line[1]));
            }else if(line[0].equals("set")){
                set(manipulatedValue, line[1], line[2]);
            }
            input.close();
        } catch (FileNotFoundException f){
            System.out.println("Error the file has not been found.");
            f.printStackTrace();
        }
        System.out.println(get(72, 6));
        System.out.println(set(5, 1, 1));
        System.out.println(comp(4, 1));
    }
}
